src/ComposerConstraint.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Drupal\lightning_dev;
+
+class ComposerConstraint {
+
+  /**
+   * E.g. '^2.8 || ^3.0'.
+   *
+   * @var string
+   */
+  private $constraint = '';
+
+  /**
+   * @param string $constraint
+   */
+  public function __construct($constraint) {
```

Comment:
All methods in this class need a one-line description, plus a description of each parameter, and return value. See the coding standards on drupal.org: https://www.drupal.org/docs/develop/coding-standards/api-documentation-and-comment-standards